### PR TITLE
packages/external-dns: Service account annotations

### LINKED ISSUE #2863: Add possibility to annotate the service account of ExternalDNS
## Feature Request

This feature would allow developers and operators to use ExternalDNS on GKE clusters that have workload identity enabled.
[Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) allows workloads in your GKE clusters to impersonate Identity and Access Management (IAM) service accounts to access Google Cloud services.

## Describe alternatives you've considered

An annotation that is manually added to the service account of ExternalDNS will eventually be overwritten by subsequent reconciliation runs.

### PR DESCRIPTION
**Pull request by TYsewyn**
## What this PR does / why we need it
This commit adds support for workload identity when running ExternalDNS on GKE clusters.

## Details for the Release Notes (PLEASE PROVIDE)

```release-note
ExternalDNS can now be used on GKE clusters that have workload identity enabled.
```

## Which issue(s) this PR fixes
Closes #2863

## Describe testing done for PR
Added a set of unit tests to verify the ytt templates.

### CODE REVIEW

**Review by christianang — CHANGES_REQUESTED** [review]
Thanks for making the PR. And thanks for writing unit tests! 🚀 

I left some comments, mostly nits. Once those are addressed I'll merge it.

File: addons/packages/external-dns/0.10.0/bundle/config/values.yaml [hunk 1]
```diff
@@ -29,8 +29,13 @@ deployment:
   #! Environment variables to pass to external-dns
   env: []
   #! Security context of the external-dns container
-  securityContext: {}
+  securityContext:
```
[the remark this answers is not in the document]
Comment (TYsewyn) [reply addons/packages/external-dns/0.10.0/bundle/config/values.yaml:32]: When I put the curly braces back and run the unit tests I get the following error:
```
        ytt: Error: Overlaying data values (in following order: values.yaml, stdin.yml):
          Document on line stdin.yml:2:
            Map item (key 'deployment') on line stdin.yml:3:
              Map item (key 'securityContext') on line stdin.yml:8:
                Map item (key 'runAsUser') on line stdin.yml:9:
                  Expected number of matched nodes to be 1, but was 0
```

[the remark this answers is not in the document]
Comment (christianang) [reply addons/packages/external-dns/0.10.0/bundle/config/values.yaml:32]: I see, if it is necessary for the tests to run then lets leave it as is.

**Review by christianang — APPROVED** [review]

**Comment by tylerschultz** [comment]
Thanks for your contribution

**Pull request merged**